src/watertap_contrib/seto/analysis/net_metering/PV_RO.py

```diff
@@ -0,0 +1,457 @@
+        >= m.fs.treatment.min_flux
+    )
+
+
+def add_costing(m):
+    treatment = m.fs.treatment
+    energy = m.fs.energy
+    treatment.costing = TreatmentCosting()
+    energy.costing = EnergyCosting()
+
+    m.db.get_unit_operation_parameters("photovoltaic")
+    energy.pv.load_parameters_from_database(use_default_removal=True)
+
+    energy.pv.costing = UnitModelCostingBlock(flowsheet_costing_block=energy.costing)
+    treatment.ro.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=treatment.costing
+    )
+    treatment.erd.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=treatment.costing
+    )
+    treatment.p1.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=treatment.costing
+    )
+
+    treatment.costing.cost_process()
+    energy.costing.cost_process()
+
+    m.fs.sys_costing = SETOSystemCosting()
+    m.fs.sys_costing.add_LCOW(treatment.product.properties[0].flow_vol)
+    m.fs.sys_costing.add_specific_electric_energy_consumption(
+        treatment.product.properties[0].flow_vol
+    )
+
+    treatment.costing.initialize()
+    energy.costing.initialize()
+
+
+def fix_pv_costing(m):
+    m.fs.energy.pv.costing.system_capacity.fix(0)
+    m.fs.energy.pv.costing.annual_generation.fix(0)
+    m.fs.energy.pv.costing.land_area.fix(0)
+
+
+def fix_treatment_global_params(m):
+    m.fs.treatment.costing.factor_total_investment.fix(1)
+    m.fs.treatment.costing.factor_maintenance_labor_chemical.fix(0)
+
+
+def size_pv(m):
+    desired_pv_size = (
+        m.fs.treatment.costing.aggregate_flow_electricity()
+        * m.fs.energy.pv.oversize_factor
+    )
+    cash_model_kwargs = {"om_fixed": 1e4, "om_production": 20}
+    m.pysam.run_pv_single_owner(
+        desired_size=desired_pv_size, cash_model_kwargs=cash_model_kwargs
+    )
+    m.fs.sys_costing.add_LCOE()
+
+
+def fix_pysam_costing(m):
+    tech_model = m.pysam.tech_model
+    cash_model = m.pysam.cash_model
+
+    avg_gen = np.mean(m.pysam.hourly_energy)
+    m.fs.energy.pv.electricity.fix(-1 * avg_gen)
+
+    annual_gen = pyunits.convert(
+        (m.pysam.annual_energy * pyunits.kWh), to_units=pyunits.MWh
+    )()
+    land_area = cash_model.LandLease.land_area * pyunits.acres
+
+    m.fs.energy.pv.costing.land_area.fix(land_area)
+    m.fs.energy.costing.photovoltaic.fixed_operating_by_capacity.fix(
+        cash_model.SystemCosts.om_capacity[0]
+    )
+    m.fs.energy.pv.costing.system_capacity.fix(m.pysam.nameplate_dc * 1000)
+    m.fs.energy.costing.photovoltaic.variable_operating_by_generation.fix(
+        cash_model.SystemCosts.om_production[0]
+    )
+    m.fs.energy.pv.costing.annual_generation.fix(annual_gen)
+    m.fs.energy.costing.factor_maintenance_labor_chemical.fix(0)
+
+
+def solve(m, solver=None, tee=False, check_termination=True):
+    if solver is None:
+        solver = get_solver()
+    results = solver.solve(m, tee=tee)
+    size_pv(m)
+    fix_pysam_costing(m)
+    results = solver.solve(m, tee=tee)
```

Comment:
Have you done any comparisons with the SAM UI to ensure your dependent parameters for costing and the DC-to-AC ratio update as expected in this Python model, and that the results agree? If not, I would highly recommend.